ref-farming/src/actions_of_reward.rs

```diff
@@ -153,6 +161,59 @@ fn claim_user_reward_from_farm(
 }
 
 impl Contract {
+    
+    /// if token was given, then just withdraw that token, or withdraw all tokens
+    pub(crate) fn internal_withdraw_all_rewards(&mut self, farmer_id: &AccountId, token: Option<&AccountId>) {
+        
+        let view_farmer = self.get_farmer(&farmer_id);
+        let tokens = {
+            if let Some(token) = token {
+                vec![token]
+            } else {
+                view_farmer.get_ref().rewards.keys().collect::<Vec<_>>()
+            }
+        };
+        
+        let mut farmer = self.get_farmer(&farmer_id);
```

Comment:
Don't need to get `view_farmer` if you still going to get `mut farmer`. It's double state read

Reply:
Refactored, and now we  have only one state read.